source/sgp_mode/SGPSymbiont.h

```diff
@@ -83,19 +83,17 @@ class SGPSymbiont : public Symbiont {
    * Purpose: To set a symbiont's host
    */
   void SetHost(emp::Ptr<Organism> host) {
-    if (!my_host) {
-      cpu.state.used_resources.Delete();
-      cpu.state.shared_available_dependencies.Delete();
-    }
     Symbiont::SetHost(host);
-    cpu.state.used_resources =
+    if (my_host) {
+      cpu.state.used_resources =
```

Comment:
Won't we have a memory leak with the FLS's previous used_resources and shared_available_dependencies once these get set to the hosts?